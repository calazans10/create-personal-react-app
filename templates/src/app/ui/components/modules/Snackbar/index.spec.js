import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Snackbar } from './index';

describe('Snackbar', () => {
  it('renders without crashing', () => {
    const props = {
      type: 'error',
      message: 'A message',
      isVisible: true,
      onHideSnackbar: jest.fn(),
    };
    const wrapper = shallow(<Snackbar {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should handle the click event', () => {
    const props = {
      type: 'error',
      message: 'A message',
      isVisible: true,
      onHideSnackbar: jest.fn(),
    };

    const wrapper = shallow(<Snackbar {...props} />);
    wrapper.find('.snackbar__close').simulate('click');
    expect(props.onHideSnackbar).toHaveBeenCalledTimes(1);
  });
});
