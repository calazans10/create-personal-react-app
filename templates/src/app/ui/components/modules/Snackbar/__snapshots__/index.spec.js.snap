// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar renders without crashing 1`] = `
<div
  className="snackbar snackbar--error snackbar--visible"
>
  <div
    className="snackbar__text"
  >
    A message
  </div>
  <button
    aria-label="Close"
    className="snackbar__close"
    onClick={[MockFunction]}
    type="button"
  />
</div>
`;
